src/main/resources/avro/bdg.avdl

```diff
@@ -57,8 +57,55 @@ record Contig {
   union { null, string } species = null;
 }
 
+record Sequence {
+  union { null, string } bases;
+  union { null, string } qualities;
+}
+
+/**
+   The DNA fragment that is was targeted by the sequencer, resulting in
+   one or more reads.
+*/
+record Fragment {
+  /**
+     The name of this Fragment.
+  */
+  union { null, string } readName = null;
+
+  union { null, string } instrument;
+	union { null, string } runId;
+  /**
+     The following are Illumina specific, is it a problem?
+   */
+  union { null, string } flowcellId;
+  union { null, int} lane = null;
+  union { null, int} tile = null;
+  union { null, int} surface = null;
+  union { null, int} xPosition = null;
+  union { null, int} yPosition = null;
+
+  /**
+    Fragment's insert size derived from alignment, if the reads have been
+    aligned.
+   */
+  union { null, int} fragmentSize = null;
+
+  /**
+   The sequences read from this fragment.
+   */
+  array<Sequence> sequences = [];
+  array<AlignmentRecord> alignments = [];
```

Comment:
`sequences` are the original sequences observed by the sequencer.  `alignments` are the alignments of those original sequences, which themselves contain a `sequence` and a `qual` field.

The reason for the duplication is that the sequence in the alignment record is transformed into a canonical format, as in SAM, where the sequence (and the base qualities) is always represented on the forward strand.  Note that this canonical format could be relatively easily reconstructed at runtime from the original sequence plus the alignment information; storing the sequence in the alignment record is thus a matter of convenience and backwards compatibility, though in principle they could easily be dropped from the AlignmentRecord.